adafruit_aw9523.py

```diff
@@ -0,0 +1,251 @@
+    """Digital input/output of the AW9523.  The interface is exactly the
+    same as the digitalio.DigitalInOut class, however:
+
+      * AW9523 family does not support pull-up or -down resistors
+
+    Exceptions will be thrown when attempting to set unsupported pull
+    configurations.
+    """
+
+    def __init__(self, pin_number, aw):
+        """Specify the pin number of the AW9523 0..15, and instance."""
+        self._pin = pin_number
+        self._aw = aw
+
+    # kwargs in switch functions below are _necessary_ for compatibility
+    # with DigitalInout class (which allows specifying pull, etc. which
+    # is unused by this class).  Do not remove them, instead turn off pylint
+    # in this case.
+    # pylint: disable=unused-argument
+    def switch_to_output(self, value=False, **kwargs):
+        """Switch the pin state to a digital output with the provided starting
+        value (True/False for high or low, default is False/low).
+        """
+        self.direction = digitalio.Direction.OUTPUT
+        self.value = value
+
+    def switch_to_input(self, pull=None, **kwargs):
+        """Switch the pin state to a digital input with the provided starting
+        pull-up resistor state (optional, no pull-up by default) and input polarity.  Note that
+        pull-down resistors are NOT supported!
+        """
+        self.direction = digitalio.Direction.INPUT
+        self.pull = pull
+
+    # pylint: enable=unused-argument
+
+    @property
+    def value(self):
+        """The value of the pin, either True for high or False for
+        low.  Note you must configure as an output or input appropriately
+        before reading and writing this value.
+        """
+        return _get_bit(self._aw.inputs, self._pin)
+
+    @value.setter
+    def value(self, val):
+        if val:
+            self._aw.outputs = _enable_bit(self._aw.outputs, self._pin)
+        else:
+            self._aw.outputs = _clear_bit(self._aw.outputs, self._pin)
+
+    @property
+    def direction(self):
+        """The direction of the pin, either True for an input or
+        False for an output.
+        """
+        if _get_bit(self._aw.direction, self._pin):
+            return digitalio.Direction.INPUT
+        return digitalio.Direction.OUTPUT
+
+    @direction.setter
+    def direction(self, val):
+        if val == digitalio.Direction.INPUT:
+            self._aw.directions = _clear_bit(self._aw.directions, self._pin)
+
+        elif val == digitalio.Direction.OUTPUT:
+            self._aw.directions = _enable_bit(self._aw.directions, self._pin)
+        else:
+            raise ValueError("Expected INPUT or OUTPUT direction!")
+
+    @property
+    def pull(self):
+        """
+        Pull-down resistors are NOT supported!
+        """
+        raise NotImplementedError("Pull-up/pull-down resistors not supported.")
+
+    @pull.setter
+    def pull(self, val):
+        raise NotImplementedError("Pull-up/pull-down resistors not supported.")
```

Comment:
With the above implementation, does switch_to_input() work?  It looks like to me the unconditional assignment of the `pull` property in `switch_to_input` will throw an exception.

If that's the case, maybe the implementation could look more like:
```

    @property
    def pull(self):
        """
        Pull-down resistors are NOT supported!
        """
        return None

    @pull.setter
    def pull(self, val):
        if val is not None:
            raise NotImplementedError("Pull-up/pull-down resistors not supported.")
```

Reply:
oh yah... it does. let me fixie!